Source/WebCore/platform/mediarecorder/MediaRecorderPrivateEncoder.cpp

```diff
@@ -411,20 +414,19 @@ void MediaRecorderPrivateEncoder::audioSamplesAvailable(const MediaTime& time, s
 
 void MediaRecorderPrivateEncoder::appendVideoFrame(VideoFrame& frame)
 {
-    if (m_isStopped)
+    if (m_isStopped || m_isPaused)
         return;
 
-    queueSingleton().dispatch([weakThis = ThreadSafeWeakPtr { *this }, this, frame = Ref { frame }, time = MediaTime(m_lastEnqueuedAudioTimeUs.load(), 1000000)]() mutable {
-        assertIsCurrent(queueSingleton());
-        if (RefPtr protectedThis = weakThis.get()) {
-            if (!hasAudio() && !m_resumeWallTime)
-                m_resumeWallTime = MonotonicTime::now();
+    if (!hasAudio() && !m_resumeWallTime.load())
+        m_resumeWallTime = MonotonicTime::now();
+
+    queueSingleton().dispatch([weakThis = ThreadSafeWeakPtr { *this }, this, frame = Ref { frame }, time = currentTime()]() mutable {
+        if (RefPtr protectedThis = weakThis.get())
             appendVideoFrame(time, WTFMove(frame));
-        }
     });
 }
 
-void MediaRecorderPrivateEncoder::appendVideoFrame(const MediaTime& audioTime, Ref<VideoFrame>&& frame)
+void MediaRecorderPrivateEncoder::appendVideoFrame(const MediaTime& time, Ref<VideoFrame>&& frame)
```

Comment:
```suggestion
void MediaRecorderPrivateEncoder::appendVideoFrame(MediaTime sampleTime, Ref<VideoFrame>&& frame)
```
This will remove the need to redeclare `sampleTime`.